tests/test_price_parsing.py

```diff
@@ -1986,3 +1988,19 @@ def test_parsing(example: Example):
 )
 def test_price_amount_float(amount, amount_float):
     assert Price(amount, None, None).amount_float == amount_float
+
+
+@pytest.mark.parametrize(
+    "price, result",
+    [
+        ('10.04.2004', datetime(2004, 4, 10, 0, 0)),
+        ('July, 2004', datetime(2004, 7, 1, 0, 0)),
+        ('Jul, 2004', datetime(2004, 7, 1, 0, 0)),
+        ('200', None),
+        ('2004', None),
+        (2004, None),
+        (10.2014, None),
+    ]
+)
+def test_date_format(price, result):
+    assert date_format(price) == result
```

Comment:
It’s great to have tests for the function, but there should also be general tests checking input strings against resulting prices.

Reply:
There are tests in general tests:

https://github.com/scrapinghub/price-parser/blob/master/tests/test_price_parsing.py#L1941
https://github.com/scrapinghub/price-parser/blob/master/tests/test_price_parsing.py#L1944